File: dart/lib/concepts/input_output.dart
import 'dart:io';

Future<void> demonstrateIO() async {
  print('\n=== Input and Output ===');

  // Output
  stdout.write('Enter your name: ');

  // Input
  String? input = stdin.readLineSync();

  // Using the input
  print('Hello, $input!');

  // Using string interpolation
  print('You entered ${input?.length ?? 0} characters');
}

File: pascal/concepts/input_output.pas
unit InputOutput;

interface

procedure DemonstrateIO;

implementation

procedure DemonstrateIO;
var
  UserInput: string;
begin
  WriteLn;
  WriteLn('=== Input and Output ===');
  
  { Output }
  Write('Enter your name: ');
  
  { Input }
  ReadLn(UserInput);
  
  { Using the input }
  WriteLn('Hello, ', UserInput, '!');
  WriteLn('You entered ', Length(UserInput), ' characters');
end;

end.